src/Symfony/Component/TypeInfo/Specification/IsA.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\Specification;
+
+use Symfony\Component\TypeInfo\Type;
+use Symfony\Component\TypeInfo\TypeIdentifier;
+use Symfony\Component\TypeInfo\Type\CollectionType;
+use Symfony\Component\TypeInfo\Type\GenericType;
+use Symfony\Component\TypeInfo\Type\IntersectionType;
+use Symfony\Component\TypeInfo\Type\ObjectType;
+use Symfony\Component\TypeInfo\Type\UnionType;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental
+ */
+final readonly class IsA
+{
+    /**
+     * @param TypeIdentifier|class-string $expected
+     */
+    public function __construct(
+        private TypeIdentifier|string $expected,
+    ) {
+    }
+
+    public function isSatifiedBy(Type $type): bool
```

Comment:
There is a big drawback of this proposal: the implementation of this method assumes that it knows all the possible implementations of Type (as it has to handle them in different way). This assumption is wrong (nothing forbids someone to create another class extending Type)

Reply:
Indeed, you were totally right, I moved back to the "abstract class" methods but in a way that ignores the concrete implementations.